homeassistant/components/climacell/sensor.py

```diff
@@ -99,17 +81,11 @@ def extra_state_attributes(self) -> Mapping[str, Any] | None:
     @property
     def unit_of_measurement(self) -> str | None:
         """Return the unit of measurement."""
-        if CONF_UNIT_OF_MEASUREMENT in self.sensor_type:
-            return self.sensor_type[CONF_UNIT_OF_MEASUREMENT]
-
-        if (
-            CONF_UNIT_SYSTEM_IMPERIAL in self.sensor_type
-            and CONF_UNIT_SYSTEM_METRIC in self.sensor_type
-        ):
+        if self.sensor_type.unit_imperial is not None:
```

Comment:
Why not do just:
```py
        if self.hass.config.units.is_metric:
            return self.sensor_type.unit_metric
        return self.sensor_type.unit_imperial
```
If `unit_imperial` is None, `unit_metric` will also be None.

Reply:
this was a holdover from the old logic. Updated, but not all sensors have units so we still need to have the `None` guard

Reply:
Don't you add `None` as default values with the dataclass?

Reply:
good point 🤦🏾